drivers/slipdev/slipdev.c

```diff
@@ -50,26 +50,87 @@ static void _slip_rx_cb(void *arg, uint8_t byte)
 {
     slipdev_t *dev = arg;
 
-    if (IS_USED(MODULE_SLIPDEV_STDIO)) {
-        if (dev->state == SLIPDEV_STATE_STDIN) {
+    switch (dev->state) {
+#if IS_USED(MODULE_SLIPDEV_STDIO)
+    case SLIPDEV_STATE_STDIN:
+        switch (byte) {
+        case SLIPDEV_ESC:
+            dev->state = SLIPDEV_STATE_STDIN_ESC;
+            break;
+        case SLIPDEV_END:
+            dev->state = SLIPDEV_STATE_NONE;
+            /* fall-through */
+        default:
             isrpipe_write_one(&slipdev_stdio_isrpipe, byte);
-            goto check_end;
+            break;
         }
-        else if ((byte == SLIPDEV_STDIO_START) &&
-            (dev->config.uart == STDIO_UART_DEV) &&
-            (dev->state == SLIPDEV_STATE_NONE)) {
+        return;
+    case SLIPDEV_STATE_STDIN_ESC:
+        switch (byte) {
+        case SLIPDEV_END_ESC:
+            byte = SLIPDEV_END;
+            break;
+        case SLIPDEV_ESC_ESC:
+            byte = SLIPDEV_ESC;
+            break;
+        }
+        dev->state = SLIPDEV_STATE_STDIN;
+        isrpipe_write_one(&slipdev_stdio_isrpipe, byte);
```

Comment:
escape bytes may need to be copied to the stdio-pipe

Reply:
Why?

Reply:
if not SLIPDEV_END bytes are unescape and the unescapend ones are then triggering end of input in stdio_read

Reply:
180a449c7a8c48f9c9183f6b30c5a8feb0c567a7 should take care of it